cpu/stm32_common/dist/clk_conf/clk_conf.c

```diff
@@ -0,0 +1,483 @@
+        else if (cfg->has_pll_sai && (cfg->has_alt_48MHz & ALT_48MHZ_SAI)) {
+            printf("PLL SAI...");
+            use_alt_48MHz = true;
+            if ((cfg->has_alt_48MHz & ALT_48MHZ_P) &&
+                    (pll_sai_p_out == 0 || pll_sai_p_out == 48000000U)) {
+                puts("P");
+                pll_sai_p_out = 48000000U;
+            }
+            else if (!(cfg->has_alt_48MHz & ALT_48MHZ_P) &&
+                     (pll_sai_q_out == 0 || pll_sai_q_out == 48000000U)) {
+                puts("Q");
+                pll_sai_q_out = 48000000U;
+            }
+            else {
+                if (cfg->has_alt_48MHz & ALT_48MHZ_P) {
+                    printf("Invalid PLL SAI P output freq: %u\n", pll_sai_p_out);
+                } else {
+                    printf("Invalid PLL SAI Q output freq: %u\n", pll_sai_q_out);
+                }
+                return 1;
+            }
+        }
+        else {
+            puts("No other source available");
+            return 1;
+        }
+        break;
+    default:
+        break;
+    }
+
+    /* PLL I2S */
+    if (pll_i2s_p_out || pll_i2s_q_out) {
+        unsigned *_m;
+        unsigned _in;
+        if (cfg->has_pll_i2s_m) {
+            _m = &m_i2s;
+        }
+        else {
+            _m = &m;
+        }
+        if (cfg->has_pll_i2s_alt_input && pll_i2s_input) {
+            _in = pll_i2s_input;
+        }
+        else {
+            _in = pll_in;
+        }
+        if (compute_pll(&cfg->pll, _in, pll_i2s_p_out, pll_i2s_q_out, 0,
+                        _m, &n_i2s, &p_i2s, &q_i2s, &r_i2s) != 0) {
+            puts("Unable to compute 48MHz output using PLL I2S");
+            return 1;
+        }
+    }
+
+    /* PLL SAI */
+    if (pll_sai_p_out || pll_sai_q_out) {
+        if (compute_pll(&cfg->pll, pll_in, pll_sai_p_out, pll_sai_q_out, 0,
+                        &m_sai, &n_sai, &p_sai, &q_sai, &r_sai) != 0) {
+            puts("Unable to compute 48MHz output using PLL I2S");
+            return 1;
+        }
+        if (!cfg->has_pll_sai_m && m != m_sai) {
+            m = m_sai;
+            DEBUG("Retry to compute main PLL with M=%u\n", m);
+            if (compute_pll(&cfg->pll, pll_in, coreclock, clock_48MHz, 0,
+                            &m, &n, &p, &q, &r) < 0) {
+                puts("Unable to compute 48MHz output using PLL I2S");
+                return 1;
+            }
+        }
+    }
+
+    /* APB prescalers */
+    unsigned apb1_pre;
+    unsigned apb2_pre;
+
+    for (apb1_pre = 1; apb1_pre <= 16; apb1_pre <<= 1) {
+        if (coreclock / apb1_pre <= cfg->max_apb1) {
+            break;
+        }
+    }
+    for (apb2_pre = 1; apb2_pre <= 16; apb2_pre <<= 1) {
+        if (coreclock / apb2_pre <= cfg->max_apb2) {
+            break;
+        }
+    }
+
+
+    /* Print constants */
+    puts("==============================================================");
+    puts("Please copy the following code into your board's periph_conf.h");
```

Comment:
please make it somehow possible to do this automatically. Either by using stderr for all other output, or by (maybe optionally) writing the actual code into a file.

Reply:
(we can do that as followup. @haukepetersen feel free to dismiss my review if you're happy.)

Reply:
Used stderr for evrything but code. Is it OK?

Reply:
Yes! thanks!